### DISCUSSION TITLE
`ColumnDataSource.patch()` logic

### DISCUSSION THREAD

**Discussion by mattpap** [bokeh/bokeh#11299] [Internals and design (Q&A)]
@bryevdv, the first `isinstance()` check makes the last one impossible:

https://github.com/bokeh/bokeh/blob/cfd5a79b2a3dea7d2ce9405e1c6693f0244ebec9/bokeh/models/sources.py#L663-L678

Should that be removed or else what should be the correct logic here?

**Comment by bryevdv** [comment]
I think maybe that `elif` can just be deleted. For mulit-index into sub-arrays I don't think we ever had any intention of allowing one patch to straddle multiple sub-arrays, i.e. the first index should be an int index into the column (to access and patch one sub-array, possibly via slices in the sub-indexes)

For 1-d columns, the first index *can* be a slice, and that is handled above, so this is probably just copy-pasta.
